ChatMessageHistory Readme and example files (#24)
averikitsch/langchaingo: examples/alloydb-chat-message-history-example/README.md

Reviewer: can you update this directory to google-alloydb-chat-message-history-example

# Google AlloyDB Chat Message History Example

> This example demonstrates how to use [AlloyDB for Postgres](https://cloud.google.com/products/alloydb) for making a Chat Message History with LangChain in Go.

Reviewer 2: [](url)
```suggestion
This example demonstrates how to use [AlloyDB for Postgres](https://cloud.google.com/products/alloydb) as a backend for the ChatMessageHistory for LangChain in Go.
```

1. **Creates an AlloyDB Chat Message History:**
   - Initializes the `alloydb.PostgresEngine` object to establish a connection to the AlloyDB database.
   - Creates a new table to store the chat message history if it doesn't exist.
   - Initializes an `alloydb.ChatMessageHistory` object, which provides methods to store, retrieve, and clear message contents with a specific session ID.

2. **Add Single Messages:**
   - Creates individual AI and Human messages and stores them in the chat message history.
   - Prints the stored messages to the console.

3. **Add Multiple Messages:**
   - Creates multiple messages of different types (AI and Human) and stores them all at once in the chat message history.
   - Prints the stored messages to the console.

4. **Overwrite Messages:**
   - Clears the existing messages and then stores a new set of messages.
   - Prints the updated list of stored messages.

5. **Clear All Messages:**
   - Clears all messages stored for the current session.

## How to Run the Example

1. Set the following environment variables. Your AlloyDB values can be found in the [Google Cloud Console](https://console.cloud.google.com/alloydb/clusters)):
   ```

>    export ALLOYDB_USERNAME=<your user>
>    export ALLOYDB_PASSWORD=<your password>
>    export ALLOYDB_DATABASE=<your database>
>    export PROJECT_ID=<your project Id>
>    export ALLOYDB_REGION=<your region>
>    export ALLOYDB_INSTANCE=<your instance>
>    export ALLOYDB_CLUSTER=<your cluster>
>    export ALLOYDB_TABLE=<your tablename>
>    export ALLOYDB_SESSION_ID=<your sessionID>
>    export ALLOYDB_SCHEMA=<your schemaName>
> 

Reviewer: This order is a little more intuitive for users
```suggestion
   export PROJECT_ID=<your project Id>
   export ALLOYDB_USERNAME=<your user>
   export ALLOYDB_PASSWORD=<your password>
   export ALLOYDB_REGION=<your region>
   export ALLOYDB_CLUSTER=<your cluster>
   export ALLOYDB_INSTANCE=<your instance>
   export ALLOYDB_DATABASE=<your database>
   export ALLOYDB_TABLE=<your tablename>
   export ALLOYDB_SESSION_ID=<your sessionID>
```
